File: Task-5/DiamondPattern.cpp
/* Diamond Pattern 15
        *
      *   *
    *       *
  *           *
*               *
  *           *
    *       *
      *   *
        *
*/
#include <iostream>
using namespace std;

int main()
{
    int n;
    cin>>n;       //Total number of rows in the diamond

    for(int i=1;i<=n/2+1;i++)       //printing the upper half of the diamond
    {
        for(int j=1;j<=(n/2+1-i)*2;j++)
            cout<<' ';
        if(i!=1)
            cout<<'*';
        for(int k=1;k<=(i-1)*4-1;k++)
            cout<<' ';
        cout<<'*'<<endl;
    }

    for(int i=1;i<=n/2;i++)        //printing the lower half of the diamond
    {
        for(int j=1;j<=i*2;j++)
            cout<<' ';
        if(i!=n/2)
            cout<<'*';
        for(int k=1;k<=(n/2-i)*4-1;k++)
            cout<<' ';
        cout<<'*'<<endl;
    }
    
    return 0;
}

File: Task-5/AlphabetsPattern/DiamondPattern.java
package AlphabetsPattern;
/* Diamond Pattern
        A
      B   B
    C       C
  D           D
E               E
  D           D
    C       C
      B   B
        A
*/
public class DiamondPattern {
    static private void printSpaces(int n){
        for(int i = 0 ; i < n; i ++) System.out.print(" ");
    }
    public static void main(String[] args) {
        int n = 5;
            
        //UPPER half
        for(int row = 1 ; row <= n ; row++){
            char base = (char)('A' + row - 1);
            printSpaces(2 * (n - row));
            System.out.print(base);

            if(row > 1){
                printSpaces(4 * row - 5);
                System.out.print(base);
            }
            System.out.println();
        }
            
        // LOWER half
        for(int row = n - 1 ; row >= 1 ; row --){
            char base = (char)('A' + row - 1);
            printSpaces(2 * (n - row));
            System.out.print(base);

            if(row > 1){
                printSpaces(4 * row - 5);
                System.out.print(base);
            }
            System.out.println();
        }
    }
}
